fe/fe-core/src/main/java/com/starrocks/sql/ast/MapExpr.java

```diff
@@ -60,12 +80,39 @@ public boolean isExplicitType() {
     @Override
     protected String toSqlImpl() {
         StringBuilder sb = new StringBuilder();
-        sb.append('[');
-        sb.append(children.stream().map(Expr::toSql).collect(Collectors.joining(",")));
-        sb.append(']');
+        if (this.explicitType) {
+            sb.append(this.type.toSql());
+        }
```

Comment:
```suggestion
```

don't output type info in explain

Reply:
but array do it.

Reply:
array will remove it later

Reply:
we can do that on map together to keep they are the same.

Reply:
I was updated array, we can print type in `explain verbose`, but don't print in `explain`, it's too complex for user?

Reply:
deleted